extension/src/SymbolReference/SymbolReferenceReader.ts

```diff
@@ -0,0 +1,263 @@
+        throw new Error(`Unexpected content length in '${appFilePath}'`);
+    }
+
+    let buffer = Buffer.from(view.getBytes(dataLength, metadataSize, true));
+
+    let zip = new AdmZip(buffer);
+    let zipEntries = zip.getEntries(); // an array of ZipEntry records
+    if (loadSymbols) {
+        symbolReference = getZipEntryContentOrEmpty(zipEntries, "SymbolReference.json");
+        symbolReference = symbolReference.replace(/\0/g, ''); // Trailing NULL characters seems to be common...
+    }
+    manifest = getZipEntryContentOrEmpty(zipEntries, "NavxManifest.xml");
+
+    return { symbolReference: symbolReference, manifest: manifest, packageId: packageId };
+}
+
+function byteArrayToGuid(byteArray: number[]): string {
+    // reverse first four bytes, and join with following two reversed, joined with following two reversed, joined with rest of the bytes
+    byteArray = (byteArray.slice(0, 4).reverse()).concat(byteArray.slice(4, 6).reverse()).concat(byteArray.slice(6, 8).reverse()).concat(byteArray.slice(8));
+
+    let guidValue = byteArray.map(function (item) {
+        // return hex value with "0" padding
+        return ('00' + item.toString(16).toUpperCase()).substr(-2, 2);
+    }).join('');
+    guidValue = `${guidValue.substr(0, 8)}-${guidValue.substr(8, 4)}-${guidValue.substr(12, 4)}-${guidValue.substr(16, 4)}-${guidValue.substr(20)}`;
+    return guidValue.toLowerCase();
+}
+
+function getZipEntryContentOrEmpty(zipEntries: AdmZip.IZipEntry[], fileName: string) {
+    let zipEntry = zipEntries.filter(zipEntry => zipEntry.name === fileName)[0];
+    if (isNullOrUndefined(zipEntry)) {
+        return '';
+    }
+    let fileContent = zipEntry.getData().toString('utf8');
+    if (fileContent.charCodeAt(0) === 0xFEFF) {
+        // Remove BOM
+        fileContent = fileContent.substr(1);
+    }
+    return fileContent;
+}
+export function getAppPackage(appFilePath: string, loadSymbols: boolean = true) {
+    let appFileContent = getAppFileContent(appFilePath);
+    let symbols: SymbolReference;
+    const manifest: ManifestPackage = (<NavxManifest>txml.simplifyLostLess(txml.parse(appFileContent.manifest) as txml.tNode[])).Package[0];
+    let appPackage: AppPackage = new AppPackage(appFilePath, manifest.App[0]._attributes.Name, manifest.App[0]._attributes.Publisher, manifest.App[0]._attributes.Version, appFileContent.packageId, manifest);
+    if (loadSymbols) {
+        symbols = <SymbolReference>JSON.parse(appFileContent.symbolReference);
+        appPackage.symbolReference = symbols;
+    }
+    return appPackage;
+}
+
+
+
+export function getObjectsFromAppFile(appFilePath: string) {
+    const { name, publisher, version } = getAppIdentifiersFromFilename(appFilePath);
+
+    let appPackage;
+    if (SymbolReferenceCache.appInCache(name, publisher, version)) {
+        appPackage = SymbolReferenceCache.getAppPackageFromCache(name, publisher, version);
+        if (appPackage) {
+            return appPackage;
+        }
```

Comment:
The check was there because of linting warnings.
But then I ensured that it is the cache already with SymbolReferenceCache.appInCache, so the check became obsolete...